web/src/containers/ActivityDetailCostAllocateFFP.js

```diff
@@ -1,96 +1,105 @@
+  };
+
+  render() {
+    const { byYearData, costAllocation } = this.props;
 
-const ActivityDetailCostAllocateFFP = ({ aId, byYearData, updateActivity }) => (
-  <div className="mb3">
-    <h4>Federal Financial Partipation (FFP) and Cost Allocation</h4>
-    <div className="clearfix mxn1">
-      {byYearData.map(({ year, total, ffpLeft, ffpData }) => (
-        <div key={year} className="col col-12 sm-col-4 px1">
-          <div className="p2 bg-darken-1">
-            <div>{year}</div>
-            <div className="h3 bold mono">{formatMoney(total)}</div>
-            <hr />
-            <div className="flex mxn-tiny">
-              {ffpData.map(({ id, percent }) => (
-                <PercentInput
-                  key={id}
-                  name={`ffp-fed-${year}`}
-                  label={ffpDisplay[id]}
-                  wrapperClass="px-tiny"
-                  value={percent}
-                  onChange={e =>
-                    updateActivity(
-                      aId,
-                      {
-                        costFFP: { [year]: { [id]: e.target.value } }
-                      },
-                      true
-                    )
-                  }
+    return (
+      <div className="mb3">
+        <h4>{t('activities.costAllocate.ffp.title')}</h4>
+        <div className="clearfix mxn1">
+          {byYearData.map(({ year, total, totalNetOther, allocations }) => (
+            <div key={year} className="col col-12 sm-col-4 px1">
+              <div className="p2 bg-darken-1">
+                <div>{year}</div>
+                <div className="h3 bold mono">{formatMoney(total)}</div>
+                <hr />
+                <DollarInput
+                  name={`cost-allocate-other-${year}`}
+                  label={t('activities.costAllocate.ffp.labels.other')}
+                  value={costAllocation[year].other}
+                  onChange={this.handleOther(year)}
                 />
-              ))}
-            </div>
-            {ffpLeft !== 0 && (
-              <div className="mt2 p1 h6 alert alert-error">
-                {ffpLeft < 0
-                  ? `You're over by ${Math.abs(ffpLeft)}%`
-                  : `Please allocate an additional ${ffpLeft}%`}
-              </div>
-            )}
-            <hr />
-            <div className="flex mxn-tiny">
-              {ffpData.map(({ id, amount }) => (
-                <div key={id} className="col-12">
-                  <div>{ffpDisplay[id]}</div>
-                  <div className="bold mono">{formatMoney(amount)}</div>
+                <div>{t('activities.costAllocate.ffp.totalNetOther')}</div>
+                <div className="h3 bold mono">{formatMoney(totalNetOther)}</div>
+                <hr />
+                <Select
+                  name={`ffp-${year}`}
+                  label={t('activities.costAllocate.ffp.labels.fedStateSplit')}
+                  options={['90-10', '75-25', '50-50']}
+                  onChange={this.handleFFP(year)}
```

Comment:
Should there be a `value` prop here so it gets set when loading?  Okay with opening a new issue for that, though.

Reply:
whoops, good catch; will update